contrib/grpc/template.py

```diff
@@ -0,0 +1,104 @@
+pkgname = "grpc"
+pkgver = "1.65.5"
+pkgrel = 0
+build_style = "cmake"
+configure_args = [
+    "-DgRPC_ABSL_PROVIDER=package",
+    "-DgRPC_BENCHMARK_PROVIDER=package",
+    "-DgRPC_CARES_PROVIDER=package",
+    "-DgRPC_PROTOBUF_PROVIDER=package",
+    "-DgRPC_RE2_PROVIDER=package",
+    "-DgRPC_SSL_PROVIDER=package",
+    "-DgRPC_ZLIB_PROVIDER=package",
+
+    "-DBUILD_SHARED_LIBS=ON",
+    "-DgRPC_BUILD_TESTS=ON",
+    "-DgRPC_INSTALL=ON",
+    "-DgRPC_DOWNLOAD_ARCHIVES=OFF",
+]
+hostmakedepends = [
+    "cmake",
+    "ninja",
+    "pkgconf"
+]
+makedepends = [
+    "abseil-cpp-devel",
+    "benchmark-devel",
+    "c-ares-devel",
+    "openssl-devel",
+    "protobuf-devel",
+    "re2-devel",
+    "zlib-ng-compat-devel",
+]
+checkdepends = [
+    "iproute2",
+    "procps",
+    "python-pyyaml"
+    "python-six",
+    "python-twisted",
+]
+pkgdesc = "High performance, open source universal RPC framework"
+maintainer = "sonata-chen <sonatachen39@gmail.com>"
+license = "Apache-2.0"
+url = "https://grpc.io"
+source = [
+    f"https://github.com/grpc/grpc/archive/refs/tags/v{pkgver}.tar.gz",
+    "https://github.com/google/googletest/archive/2dd1c131950043a8ad5ab0d2dda0e0970596586a.tar.gz",
+]
+source_paths = [".", "third_party/googletest"]
+sha256 = [
+    "bdab9cb52e3ee703d6c09d15716bcc10a150b2e9e414182e5b375d32050dba84",
+    "31bf78bd91b96dd5e24fab3bb1d7f3f7453ccbaceec9afb86d6e4816a15ab109",
+]
+
+
+def post_build(self):
+    special_tests = [
+        # RTNETLINK answers: Operation not permitted
+        "flaky_network_test",  # disabled
+
+        # it takes 58 mins to run
+        "grpc_tool_test",  # disabled
+
+        # need flags to run, so we run them manually
+        # tools/run_tests/generated/tests.json
+        "resolve_address_using_ares_resolver_posix_test",  # --resolver ares
+        "resolve_address_using_native_resolver_posix_test",  # --resolver native
+
+
+        # E0000 00:00:1724420309.061535      12 ssl_utils.cc:577] error loading file ./build/../../src/core/tsi/test_creds/ca.pem:
+        # INTERNAL: Failed to load file: ./build/../../src/core/tsi/test_creds/ca.pem due to error(fdopen): No such file or directory
+        #
+        # test/core/http/httpcli_test_util.cc
+        # we are going to run them with modified PATH environment variable
+        "httpcli_test",
+        "httpscli_test",
+    ]
+    with self.pushd("build"):
+        for t in special_tests:
+            self.mv(f"{t}", f"{t}_")
```

Comment:
if you rerun the build step this fails so you can't do this

Reply:
This would not cause a problem, since cmake will re-link the missing executables.

Reply:
i suppose that's true